tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/CodeownersHelper.cs

```diff
@@ -0,0 +1,587 @@
+                if (serviceLabel1.Equals(dirName, StringComparison.OrdinalIgnoreCase) ||
+                    dirName.Equals(serviceLabel1.Replace("%", "").Trim(), StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            // 2: Compare service label to PR label
+            if (!string.IsNullOrEmpty(serviceLabel1) && !string.IsNullOrEmpty(PRLabel2))
+            {
+                if (string.Equals(serviceLabel1.Replace("%", "").Trim(), PRLabel2.Replace("%", "").Trim(), StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            if (!string.IsNullOrEmpty(PRLabel1) && !string.IsNullOrEmpty(serviceLabel2))
+            {
+                if (string.Equals(PRLabel1.Replace("%", "").Trim(), serviceLabel2.Replace("%", "").Trim(), StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            // 3: Compare by service label
+            if (!string.IsNullOrEmpty(serviceLabel1) && !string.IsNullOrEmpty(serviceLabel2))
+            {
+                if (string.Equals(serviceLabel1.Replace("%", "").Trim(), serviceLabel2.Replace("%", "").Trim(), StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private (string serviceLabel, string PRLabel) GetPrimaryLabel(CodeownersEntry entry)
+        {
+            var serviceLabel = entry.ServiceLabels?.FirstOrDefault(label => !string.IsNullOrWhiteSpace(label)) ?? string.Empty;
+            var PRLabel = entry.PRLabels?.FirstOrDefault(label => !string.IsNullOrWhiteSpace(label)) ?? string.Empty;
+            return (serviceLabel, PRLabel);
+        }
+
+        public string ExtractDirectoryName(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                return string.Empty;
+            }
+
+            var parts = path.Trim('/').Split('/');
+
+            return parts.LastOrDefault(part => !string.IsNullOrWhiteSpace(part)) ?? string.Empty;
+        }
+
+        public string CreateBranchName(string prefix, string identifier)
+        {
+            var normalizedIdentifier = NormalizeIdentifier(identifier);
+            return $"{prefix}-{normalizedIdentifier}";
+        }
+
+        public string NormalizeIdentifier(string input)
+        {
+            input = input
+                .Replace(" - ", "-")
+                .Replace(" ", "-")
+                .Replace("/", "-")
+                .Replace("_", "-")
+                .Replace(".", "-")
+                .Trim('-')
+                .ToLowerInvariant();
+            return Regex.Replace(input, @"[^a-zA-Z0-9\-]", "");
+        }
+
+        public List<string> AddUniqueOwners(List<string> existingOwners, List<string> ownersToAdd)
```

Comment:
```suggestion
        public List<string> AddOwners(List<string> existingOwners, List<string> ownersToAdd)
```

One can probably assume deduplication.